v2/client.go

```diff
@@ -134,19 +139,23 @@ func requestContentType(ct string) requestOption {
 	}
 }
 
-func (c *Client) buildRequest(ctx context.Context, method, uri string, opts ...requestOption) (*http.Request, error) {
+func (c *Client) buildURL(pathElements ...any) *url.URL {
```

Comment:
It occurred to me that we need to sanitize paths in more places than just the tailnet name, so I suggest we switch to this function for building our URLs, which takes care of that.